src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -129,6 +141,26 @@ public static String jenkinsJobName(BuildConfig bc) {
     return jenkinsJobName(namespace, name);
   }
 
+  // Since the token we need can change without updating the global config, we need to get it every time we do a call
+  public static void setOpenShiftOauthToken() {
+    GlobalPluginConfiguration config = GlobalConfiguration.all().get(GlobalPluginConfiguration.class);
+    if(config == null){
+       return;
+    }
+    String credentialsId = config.getCredentialsId();
+    List<OpenShiftTokenCredentials> tokens = CredentialsProvider.lookupCredentials(OpenShiftTokenCredentials.class, (ItemGroup)null, ACL.SYSTEM , (DomainRequirement)null);
+    if( tokens != null ) {
+      for (OpenShiftTokenCredentials token : tokens) {
+        if (token.getId().equals(credentialsId)) {
+          openShiftClient.getConfiguration().setOauthToken(token.getToken());
+          return;
+        }
+      }
+    }
```

Comment:
Another thought occurred to me.  If absolutely no open shift token credentials are defined, we should consider dynamically create and use a openshift token credential based on the token mounted at /var/run/secrets/kubernetes.io/serviceaccount/token

 If that mount point is not available, then move forward with the exception below